src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -291,6 +309,8 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
 
                     if (!request.HasHeaders || request.Headers.ExpectContinue != true)
                     {
+                        _canRetry = false;
```

Comment:
Should this be moved to SendRequestContentAsync so that it's also set to false when we start sending for Expect: 100-Continue?

Reply:
There's a separate spot where we handle the Expect: 100-continue case.

I actually originally had this in SendRequestContentAsync, but the comment here made me change it:

```csharp
// Start the copy from the request.  We do this here in case it synchronously throws
// an exception, e.g. StreamContent throwing for non-rewindable content, and because if
// we did it in SendRequestContentAsync, that exception would get trapped in the returned
// task... at that point, we might get stuck waiting to receive a response from the server
// that'll never come, as the server is still expecting us to send data.
_sendRequestContentTask = SendRequestContentAsync(request.Content.CopyToAsync(stream, _transportContext), stream);
```

So it seems that a synchronous exception from Content.CopyToAsync is possible here, and I want to make sure _canRetry is set to false before that exception.

That said, the current logic here doesn't seem ideal to me -- what happens if Content.CopyToAsync fails asynchronously without sending any data? Or sending partial data?  Won't that cause the same problem described here?

Reply:
> So it seems that a synchronous exception from Content.CopyToAsync is possible here, and I want to make sure _canRetry is set to false before that exception.

Makes sense.

> what happens if Content.CopyToAsync fails asynchronously without sending any data? Or sending partial data? Won't that cause the same problem described here?

Yeah, you're right, this would be similar.  I think my assumption was at that point if there's a failure it's because we failed to write to the connection, at which point the connection is bad and something else will fail and we won't get stuck.  But it's feasible there could be another reason for the failure, in which case we may sit waiting for the response from the server and the server may still be waiting for our request and we essentially deadlock until one of use decides to timeout and close the connection.

I opened https://github.com/dotnet/corefx/issues/26741.  I think there's a relatively straightforward fix.